src/components/Recipes/Recipes.jsx

```diff
@@ -81,35 +86,107 @@ export default function Recipes(props) {
     }
   }, [userCart])
 
+  // update user's products in database
+  useEffect(() => {
+    if (!props.isLoading && userProducts) {
+      const docRef = doc(db, "users", currentUser.uid)
+      updateDoc(docRef, { products: userProducts })
+        .catch(error => {
+          setError(error.message)
+      })
+    }
+  }, [userProducts])
```

Comment:
I think you need to include props.isLoading, in the list here.

Reply:
Would this happen if I try to update the database while refreshing the page? Logically I don't think I try to update the database while the page is rendering (since the user has to physically click on the button for userProducts to change), but the more I think the more I get confused about the lifecycle. Perhaps Monday will clear things up